test/standalone/Soil/soil_test_3d.jl

```diff
@@ -415,3 +415,49 @@ end
     ]) / ν < 1e-11
 
 end
+
+
+@testset "Lateral flow flag" begin
+    FT = Float32
+    zmax = FT(0)
+    zmin = FT(-1)
+    xmax = FT(1.0)
+    boxdomain = ClimaLSM.Domains.HybridBox(;
+        xlim = (zmin, zmax),
+        ylim = (zmin, zmax),
+        zlim = (zmin, zmax),
+        nelements = (100, 2, 100),
+        npolynomial = 3,
+    )
+    shell = ClimaLSM.Domains.SphericalShell(;
+        radius = FT(1),
+        height = FT(1),
+        nelements = (2, 3),
+        npolynomial = 1,
+    )
+    column = ClimaLSM.Domains.Column(; zlim = (zmin, zmax), nelements = 10)
+    for d in [boxdomain, shell]
+        f1 = ClimaCore.Fields.zeros(d.space)
+        f2 = ClimaCore.Fields.zeros(d.space)
+        dY = ClimaCore.Fields.FieldVector(; g = f1, h = f2)
+        ClimaLSM.Soil.horizontal_components!(dY, d, Val(false))
+        @test dY.g == f1
+        @test dY.h == f2
+        # If you pass true, you need additional arguments (dispatch off of model type)
```

Comment:
Should we also add a test where the additional arguments are provided and we check that the horizontal components are correctly computed?

Reply:
I think this is ok because the other tests in this file were already implicitly testing horizontal_components! in the case where they were nonzero, and those are the default now (with lateral_flow = true).